docs/mobile/events.md

```diff
@@ -21,46 +20,57 @@ application lifecycle. This event is meant to be used in conjunction with
 mobile operating system (e.g. Android, iOS).
 
-### iOS
+Payload attributes MUST be used to describe the state of the application at the
+time of the event. The following table describes the payload attributes that MUST
+be used to describe the state of the application at the time of the event.
 
-<!-- semconv ios.lifecycle.events -->
-The event name MUST be `device.app.lifecycle`.
-
-| Attribute  | Type | Description  | Examples  | Requirement Level |
-|---|---|---|---|---|
-| `ios.state` | string | This attribute represents the state the application has transitioned into at the occurrence of the event. [1] | `active` | Required |
+The `android.state` and `ios.state` attributes are mutually exclusive and MUST
+NOT be used together, each attribute MUST be used with it's corresponding
+`os.name` [resource semantic convention](/docs/resource/os.md) value. For
+example, if the `os.name` attribute is set to `android` then the
+`android.state` attribute MUST be used and the `ios.state` attribute MUST NOT
+be used. If the `os.name` attribute is set to `ios` then the `ios.state`
+attribute MUST be used and the `android.state` attribute MUST NOT be used.
 
-**[1]:** The iOS lifecycle states are defined in the [UIApplicationDelegate documentation](https://developer.apple.com/documentation/uikit/uiapplicationdelegate#1656902), and from which the `OS terminology` column values are derived.
+### Event details
 
-`ios.state` MUST be one of the following:
+<!-- semconv device.app.lifecycle(full) -->
+The event name MUST be `device.app.lifecycle`.
 
-| Value  | Description |
-|---|---|
-| `active` | The app has become `active`. Associated with UIKit notification `applicationDidBecomeActive`. |
-| `inactive` | The app is now `inactive`. Associated with UIKit notification `applicationWillResignActive`. |
-| `background` | The app is now in the background. This value is associated with UIKit notification `applicationDidEnterBackground`. |
-| `foreground` | The app is now in the foreground. This value is associated with UIKit notification `applicationWillEnterForeground`. |
-| `terminate` | The app is about to terminate. Associated with UIKit notification `applicationWillTerminate`. |
 <!-- endsemconv -->
 
-### Android
-
-<!-- semconv android.lifecycle.events -->
-The event name MUST be `device.app.lifecycle`.
-
-| Attribute  | Type | Description  | Examples  | Requirement Level |
+<!-- Manually adding the markdown table until the body definition is available in the build tools -->
+| Body Field | Type | Description  | Examples  | Requirement Level |
 |---|---|---|---|---|
-| `android.state` | string | This attribute represents the state the application has transitioned into at the occurrence of the event. [1] | `created` | Required |
+| `android.state` | string | This attribute represents the state the application has transitioned into at the occurrence of the event. [1] | `created` | See below |
```

Comment:
I think using `constraints` makes markdown worse than is could be.  With `requirement_level: conditionally_required: if and only if `os.name` is `android`` it would look better and would eliminate the need for mutual exclusivity notes above and below.

```suggestion
| `android.state` | string | This attribute represents the state the application has transitioned into at the occurrence of the event. [1] | `created` | `Conditionally Required`: if and only if `os.name` is `android`|
```